backend/src/adapters/mailer/mailer.service.ts

```diff
@@ -0,0 +1,38 @@
+import { Injectable } from '@nestjs/common';
+import { ConfigService } from '@nestjs/config';
+import { MAILER_CONFIG } from '@src/config/config';
+import { IMailerConfig } from '@src/config/mailer.config';
+import * as nodemailer from 'nodemailer';
+import * as pug from 'pug';
+
+@Injectable()
+export class MailerService {
+    private transporter;
+    private sender;
+
+    constructor(private configService: ConfigService) {
+        this.sender = this.configService.get<IMailerConfig>(MAILER_CONFIG)?.EMAIL_ADDRESS;
+        this.transporter = nodemailer.createTransport({
+            host: this.configService.get<IMailerConfig>(MAILER_CONFIG)?.MAILER_HOST,
+            port: this.configService.get<IMailerConfig>(MAILER_CONFIG)?.MAILER_PORT,
+            secure: false,
+            requireTLS: true,
+            auth: {
+                user: this.sender,
+                pass: this.configService.get<IMailerConfig>(MAILER_CONFIG)?.EMAIL_PASS,
+            }
+        });
+    }
+
+    async sendConfirmEmailLetter(recipient: string, link: string): Promise<void> {
```

Comment:
Create separate file in core folder called ConfirmEmailService which injects MailerAdapter and has method sendConfirmEmailLetter.
In MailerService leave method
```
sendEmail(recipient: string, messageConfig: Message): Promise<void> {
let message = pug.renderFile(messageConfig.templatePath, messageConfig.payload);
        ...rest
        
type MessageConfig = { templatePath: './templates/confirmEmail.pug', payload: { confirmationLink: string } }
(here we will join multiple objects (for different emails)
```